token/program-2022/src/processor.rs

```diff
@@ -312,16 +336,43 @@ impl Processor {
             return Ok(());
         }
 
-        source_account.amount = source_account
+        // self-transfer was dealt with earlier, so this *should* be safe
```

Comment:
So just confirming, self transfers with fees should not have fees collected? (Though I guess you can still test the expected fee with TransferCheckedWithFee.)
It would be nice to have a test case in transfer_fee.rs demonstrating/reiterating no fees on self-transfers.

Reply:
Correct, the idea here is that self-transfers do not accrue fees, based on the idea that self-transfers signed by a delegate don't eat until the amount delegated.  Ie, I approve 10 tokens to you, and if you self transfer them, your 10 approved tokens don't change.

That's not a great analogy though, so I don't feel strongly about this.  Maybe @mvines or @samkim-crypto have an opinion on the desired behavior?

Either way, I'll certainly add a test case for this.

Reply:
> the idea here is that self-transfers do not accrue fees

👍🏼